mgmtd/mgmt_txn.c

```diff
@@ -2093,6 +2103,19 @@ int mgmt_txn_notify_be_adapter_conn(struct mgmt_be_client_adapter *adapter,
 			return 0;
 		}
 
+		ds_ctx = mgmt_ds_get_ctx_by_id(mm, MGMTD_DS_RUNNING);
+		assert(ds_ctx);
+
+		/*
+		 * Lock the running datastore to prevent any changes while we
+		 * are initializing the backend.
+		 */
+		if (mgmt_ds_lock(ds_ctx, 0) != 0) {
```

Comment:
I agree, but I didn't want to rearrange the order of operations in this fix. We're already doing the `mgmt_txn_create_new` after getting the changes, and it can fail because of the same thing - another running transaction. So putting the locking here seems reasonable. We can rearrange the order in a separate PR.